platform/server/http/Router/Response.ts

```diff
@@ -5,18 +5,13 @@ export class KibanaResponse<T> {
   constructor(readonly status: StatusCode, readonly payload?: T) {}
 }
 
-export interface ResponseFactory {
-  ok<T extends { [key: string]: any }>(payload: T): KibanaResponse<T>;
-  accepted<T extends { [key: string]: any }>(payload: T): KibanaResponse<T>;
-  noContent(): KibanaResponse<void>;
-  badRequest<T extends Error>(err: T): KibanaResponse<T>;
-}
-
-export const responseFactory: ResponseFactory = {
+export const responseFactory = {
   ok: <T extends { [key: string]: any }>(payload: T) =>
     new KibanaResponse(200, payload),
   accepted: <T extends { [key: string]: any }>(payload: T) =>
     new KibanaResponse(202, payload),
   noContent: () => new KibanaResponse<void>(204),
   badRequest: <T extends Error>(err: T) => new KibanaResponse(400, err)
 };
+
+export type ResponseFactory = typeof responseFactory;
```

Comment:
Why this way of defining the ResponseFactory type? It's based on the type of a const that's defined prior. It makes sense, but why this way?

Reply:
There wasn't really that much thought behind this. In the previous version `ResponseFactory` was defined as an interface, but I didn't feel it was necessary to both define the type and the `responseFactory` implementation when they are located in the same file, so this felt like the "simplest" approach.